Grid/qcd/utils/GaugeGroupTwoIndex.h

```diff
@@ -108,60 +178,53 @@ class GaugeGroupTwoIndex : public GaugeGroup<ncolour, group_name> {
     }
 
     if (Index < ncolour * (ncolour - 1) / 2) {
-      baseOffDiagonal(a[Index][0], a[Index][1], eij, group_name());
+      baseOffDiagonal(a[Index][0], a[Index][1], eij);
     } else {
       baseDiagonal(Index, eij);
     }
   }
+    
+  template <class cplx>
+   static void base(int Index, iGroupMatrix<cplx> &eij, GroupName::Sp) {
+     // returns (e)^(ij)_{kl} necessary for change of base U_F -> U_R
+     assert(Index < Dimension);
+     eij = Zero();
+
+     // for the linearisation of the 2 indexes
+     static int a[ncolour * (ncolour - 1) / 2][2];  // store the a <-> i,j
+     static bool filled = false;
+     if (!filled) {
+       int counter = 0;
+       for (int i = 1; i < ncolour; i++) {
+         for (int j = 0; j < i; j++) {
+           a[counter][0] = i;
+             if (j==0 && i==ngroup+j && S==-1) {
+                 //std::cout << "skipping" << std::endl; // for Sp2n this vanishes identically.
+                 j = j+1;
+             }
+           a[counter][1] = j;
+           counter++;
+         }
+       }
+       filled = true;
+     }
+
+     if (Index < DimensionHelper<ncolour, AntiSymmetric, GroupName::Sp>::Dimension + 1) {   // +1 the singlet
+       baseOffDiagonalSp(a[Index][0], a[Index][1], eij);
```

Comment:
I am a bit confused, you suggest to not call it?

Reply:
Sorry, my bad. I meant directly having
```cpp
baseOffDiagonalSpHelper<...>::baseOffDiagonalSp(...)
```
instead of first calling `baseOffDiagonalSp` which itself consists of the single line above.

Reply:
ah yes sorry! I was looking at the dimension helper